### DISCUSSION TITLE
can you explain how to convert wavelength to XYZ

### DISCUSSION THREAD

**Discussion by gravitychen** [rafael-fuente/Diffraction-Simulations--Angular-Spectrum-Method#20] [Q&A] [answered]
Thank you very much.
can you explain how to convert wavelength to XYZ
what is the principle behind this

**Comment by rafael-fuente** [comment, marked as answer]
Hi gravitychen,

Conversion from wavelength to RGB colors according to the human color perception is implemented using [CIE color matching functions](https://en.wikipedia.org/wiki/CIE_1931_color_space#:~:text=Color%20matching%20functions,-The%20CIE%20XYZ&text=are%20the%20numerical%20description%20of,values%20X%2C%20Y%20and%20Z.) 
This is explained in the polychromatic light section in the [angular spectrum post](https://rafael-fuente.github.io/simulating-diffraction-patterns-with-the-angular-spectrum-method-and-python.html).
